arches/app/etl_modules/base_import_module.py

```diff
@@ -367,27 +370,147 @@ def start(self, request):
         return {"success": result["started"], "data": result}
 
     def write(self, request):
-        self.loadid = request.POST.get("load_id")
         self.temp_dir = os.path.join(settings.UPLOADED_FILES_DIR, "tmp", self.loadid)
         self.file_details = request.POST.get("load_details", None)
+        multiprocessing = request.POST.get("multiprocessing", False)
         result = {}
         if self.file_details:
             details = json.loads(self.file_details)
             files = details["result"]["summary"]["files"]
             summary = details["result"]["summary"]
-            use_celery_file_size_threshold_in_MB = 0.1
-            if (
-                summary["cumulative_files_size"] / 1000000
-                > use_celery_file_size_threshold_in_MB
-            ):
+            use_celery_file_size_threshold = self.config.get(
+                "celeryByteSizeLimit", 100000
+            )
+
+            if summary["cumulative_files_size"] > use_celery_file_size_threshold:
                 response = self.run_load_task_async(request, self.loadid)
             else:
                 response = self.run_load_task(
-                    self.userid, files, summary, result, self.temp_dir, self.loadid
+                    self.userid,
+                    files,
+                    summary,
+                    result,
+                    self.temp_dir,
+                    self.loadid,
+                    multiprocessing,
                 )
 
             return response
 
+    def cli(self, source):
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            read = self.read_for_cli(source)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": initiated["data"]["message"]},
+            }
+
+        if read["success"]:
+            self.request.POST.__setitem__(
+                "load_details", json.dumps({"result": read["data"]})
+            )
+            self.request.POST.__setitem__("multiprocessing", True)
+            written = self.write(self.request)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": read["data"]["message"]},
+            }
+
+        if written["success"]:
+            return {"success": True, "data": "done"}
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": written["data"]["message"]},
+            }
+
+    def read_for_cli(self, source):
```

Comment:
This method is pretty long and looks like it shares some code with the `read` method. I'm wondering if some of it could be pulled out and shared between methods.